### PR TITLE
Added DNS Chaos doc

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

**What this PR does / why we need it**:

This PR adds DNS Error and Spoof experiment documentation

**Checklist:**
-   [ ] Fixes #<issue number>
-   [x] Signed the commit for [DCO](https://github.com/litmuschaos/litmus-docs/blob/master/CONTRIBUTING.md#sign-your-work) check to be passed
-   [ ] Labelled this PR & related issue with `documentation` tag

### CODE REVIEW

File: docs/dns-error.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
+---
+id: pod-dns-error
```
Comment (ispeakc0de) [inline docs/dns-error.md]: change the file name to pod-dns-error.md?

File: docs/pod-dns-error.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+<table>
+  <tr>
+    <th> Variables </th>
+    <th> Description  </th>
+    <th> Specify In ChaosEngine </th>
+    <th> Notes </th>
+  </tr>
+  <tr>
+    <td> TARGET_CONTAINER  </td>
+    <td> Name of container which is subjected to network loss </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-error.md]:
```suggestion
    <td> Name of container which is subjected to dns-error </td>
```

File: docs/pod-dns-error.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  </tr>
+  <tr>
+    <td> TARGET_HOSTNAMES </td>
+    <td> List of the target hostnames or keywords eg. '["litmuschaos","chaosnative.com"]'</td>
+    <td> Optional </td>
+    <td> If not provided, all hostnames/domains will be targeted</td>
+  </tr> 
+  <tr>
+    <td> MATCH_SCHEME </td>
+    <td> Determines whether the dns query has to match exactly with one of the targets or can have any of the targets as substring. Can be either `exact` or `substring` </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-error.md]:
```suggestion
    <td> Determines whether the dns query has to match exactly with one of the targets or can have any of the targets as substring. Can be either <code>exact</code> or <code>substring</code> </td>
```

File: docs/pod-dns-error.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    <td> PODS_AFFECTED_PERC </td>
+    <td> The Percentage of total pods to target  </td>
+    <td> Optional </td>
+    <td> Defaults to 0 (corresponds to 1 replica), provide numeric value only </td>
+  </tr> 
+  <tr>
+    <td> CONTAINER_RUNTIME  </td>
+    <td> container runtime interface for the cluster</td>
+    <td> Optional </td>
+    <td> Defaults to docker, supported values: docker, containerd and crio for litmus and only docker for pumba LIB </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-error.md]:
```suggestion
    <td> Defaults to docker, supported values: docker, containerd and crio </td>
```

File: docs/pod-dns-error.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+    <td> SOCKET_PATH </td>
+    <td> Path of the containerd/crio/docker socket file </td>
+    <td> Optional  </td>
+    <td> Defaults to `/var/run/docker.sock` </td>
+  </tr>
+  <tr>
+    <td> LIB </td>
+    <td> The chaos lib used to inject the chaos </td>
+    <td> Optional  </td>
+    <td> Default value: litmus, supported values: pumba and litmus </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-error.md]:
```suggestion
    <td> Default value: litmus, supported values: litmus </td>
```

File: docs/pod-dns-error.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+### Watch Chaos progress
+
+- View dns failure by setting up a ping on the target domain/hostname from the affected pod, should error out.
+
+  `ping <target_domain>`
+
+### Abort/Restart the Chaos Experiment
+
+- To stop the pod-network-loss experiment immediately, either delete the ChaosEngine resource or execute the following command:
```
Comment (ispeakc0de) [inline docs/pod-dns-error.md]:
```suggestion
- To stop the pod-dns-error experiment immediately, either delete the ChaosEngine resource or execute the following command: 
```

File: docs/pod-dns-error.md [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+
+  `kubectl patch chaosengine <chaosengine-name> -n <namespace> --type merge --patch '{"spec":{"engineState":"stop"}}'` 
+
+- To restart the experiment, either re-apply the ChaosEngine YAML or execute the following command: 
+
+  `kubectl patch chaosengine <chaosengine-name> -n <namespace> --type merge --patch '{"spec":{"engineState":"active"}}'`  
+
+### Check Chaos Experiment Result
+
+- Check whether the application is resilient to the Pod Network Loss, once the experiment (job) is completed. The ChaosResult resource name is derived like this: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.
```
Comment (ispeakc0de) [inline docs/pod-dns-error.md]:
```suggestion
- Check whether the application is resilient to the Pod Dns Chaos, once the experiment (job) is completed. The ChaosResult resource name is derived like this: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.
```

File: docs/pod-dns-spoof.md [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+
+- Follow the steps in the sections below to create the chaosServiceAccount, prepare the ChaosEngine & execute the experiment.
+
+### Prepare chaosServiceAccount
+
+- Use this sample RBAC manifest to create a chaosServiceAccount in the desired (app) namespace. This example consists of the minimum necessary role permissions to execute the experiment.
+
+#### Sample Rbac Manifest
+
+[embedmd]:# (https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/generic/pod-dns-error/rbac.yaml yaml)
```
Comment (ispeakc0de) [inline docs/pod-dns-spoof.md]: do you mean pod-dns-spoof?

File: docs/pod-dns-spoof.md [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+<table>
+  <tr>
+    <th> Variables </th>
+    <th> Description  </th>
+    <th> Specify In ChaosEngine </th>
+    <th> Notes </th>
+  </tr>
+  <tr>
+    <td> TARGET_CONTAINER  </td>
+    <td> Name of container which is subjected to network loss </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-spoof.md]:
```suggestion
    <td> Name of container which is subjected to dns spoof </td>
```

File: docs/pod-dns-spoof.md [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+    <td> PODS_AFFECTED_PERC </td>
+    <td> The Percentage of total pods to target  </td>
+    <td> Optional </td>
+    <td> Defaults to 0 (corresponds to 1 replica), provide numeric value only </td>
+  </tr> 
+  <tr>
+    <td> CONTAINER_RUNTIME  </td>
+    <td> container runtime interface for the cluster</td>
+    <td> Optional </td>
+    <td> Defaults to docker, supported values: docker, containerd and crio for litmus and only docker for pumba LIB </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-spoof.md]:
```suggestion
    <td> Defaults to docker, supported values: docker, containerd and crio </td>
```

File: docs/pod-dns-spoof.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+    <td> SOCKET_PATH </td>
+    <td> Path of the containerd/crio/docker socket file </td>
+    <td> Optional  </td>
+    <td> Defaults to `/var/run/docker.sock` </td>
+  </tr>
+  <tr>
+    <td> LIB </td>
+    <td> The chaos lib used to inject the chaos </td>
+    <td> Optional  </td>
+    <td> Default value: litmus, supported values: pumba and litmus </td>
```
Comment (ispeakc0de) [inline docs/pod-dns-spoof.md]:
```suggestion
    <td> Default value: litmus, supported values: litmus </td>
```

File: docs/pod-dns-spoof.md [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+
+  `ping <target_domain>`
+
+- Ping the target hostname/domain again from the affected pod, during chaos and you should notice the ip points to the spoofed destination
+
+  `ping <target_domain>`
+
+### Abort/Restart the Chaos Experiment
+
+- To stop the pod-network-loss experiment immediately, either delete the ChaosEngine resource or execute the following command:
```
Comment (ispeakc0de) [inline docs/pod-dns-spoof.md]:
```suggestion
- To stop the pod-dns-spoof experiment immediately, either delete the ChaosEngine resource or execute the following command: 
```

File: docs/pod-dns-spoof.md [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+
+  `kubectl patch chaosengine <chaosengine-name> -n <namespace> --type merge --patch '{"spec":{"engineState":"stop"}}'` 
+
+- To restart the experiment, either re-apply the ChaosEngine YAML or execute the following command: 
+
+  `kubectl patch chaosengine <chaosengine-name> -n <namespace> --type merge --patch '{"spec":{"engineState":"active"}}'`  
+
+### Check Chaos Experiment Result
+
+- Check whether the application is resilient to the Pod Network Loss, once the experiment (job) is completed. The ChaosResult resource name is derived like this: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.
```
Comment (ispeakc0de) [inline docs/pod-dns-spoof.md]:
```suggestion
- Check whether the application is resilient to the Pod Dns Spoof, once the experiment (job) is completed. The ChaosResult resource name is derived like this: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.
```

**Review by ispeakc0de — DISMISSED** [review]

File: docs/pod-dns-error.md [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+    <th> Variables </th>
+    <th> Description  </th>
+    <th> Specify In ChaosEngine </th>
+    <th> Notes </th>
+  </tr>
+  <tr>
+    <td> TARGET_CONTAINER  </td>
+    <td> Name of container which is subjected to dns-error </td>
+    <td> Optional </td>
+    <td> Applicable for containerd & CRI-O runtime only. Even with these runtimes, if the value is not provided, it injects chaos on the first container of the pod</td>
```
Comment (ksatchit) [inline docs/pod-dns-error.md]: We can leave this empty. We now use sandbox containers for docker/containerd/crio runtime

File: docs/pod-dns-spoof.md [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+  </tr>
+  <tr>
+    <td> TOTAL_CHAOS_DURATION </td>
+    <td> The time duration for chaos insertion (seconds) </td>
+    <td> Optional </td>
+    <td> Default (60s) </td>
+  </tr>
+  <tr>
+    <td> SPOOF_MAP </td>
+    <td> Map of the target hostnames eg. '{"abc.com":"spoofabc.com"}' where key is the hostname that needs to be spoofed and value is the hostname where it will be spoofed to.</td>
```
Comment (ksatchit) [inline docs/pod-dns-spoof.md]: redirected/changed to?

**Review by ksatchit — DISMISSED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**